Adds dropdown menus for the due date (dd/mm/yyyy)

NickCampagna opened the pull request:
Adds three dropdown menus to adjust the day, month and year of the due date, using the dropdowns for status/assigned user as a template. I couldn't figure out how to get the calendar button to toggle the daypicker from showing, but left the import in in case anyone wants to take a shot at it. Currently, changing the date does not cause the "days left" string to re-calculate, and I realize that it's possible to set "impossible dates" (i.e. February 31st) but I am currently unsure how to restrict this from happening in React. All in all, it could be better, but hopefully it's decent enough for a start.

theroyalblah commented on src/components/dueDateDropdown.tsx:
```diff
@@ -0,0 +1,176 @@
+
+        // Maps through the array given and sets up the options
+        // Needs to be done in the render() function or will not produce the proper output
+        const arrayOp = this.options.map((item, i) => {
```
i is left unused here

theroyalblah commented on src/components/dueDateDropdown.tsx:
```diff
@@ -0,0 +1,176 @@
+
+        // Maps through the array given and sets up the options
+        // Needs to be done in the render() function or will not produce the proper output
+        const arrayOp = this.options.map((item, i) => {
```
same as earlier, i is left unused

theroyalblah commented on src/components/taskView.tsx:
```diff
@@ -166,6 +199,73 @@ export class TaskView extends React.Component<TaskViewProps>{
         ];
+		
+		this.yearDue = this.today.getFullYear();
+		this.years = [
```
These shouldn't really be here. They just get passed down and are constant anyhow so I think:
1. They should not be a class member variable
2. They should be a constant in dueDateDropdown

theroyalblah commented on src/components/taskView.tsx:
```diff
@@ -166,6 +199,73 @@ export class TaskView extends React.Component<TaskViewProps>{
+			{id: 11, month: 12},
+		];
+		
+		this.dayDue = this.today.getDate();
```
This is a lot, and I think we can come up with a better solution than a giant array of possible days

theroyalblah commented on src/components/dueDateDropdown.tsx:
```diff
@@ -0,0 +1,176 @@
+    yearList: IYears[];
+}
+
+export class DayDropdown extends React.Component<IDayDropdownProps, IDayState> {
```
A ton of code gets repeated here. I think a clever solution to this would involve handing some identifier down and determining how to present that day all in one class. Avoiding large copy paste jobs is always good.

theroyalblah commented on src/components/taskView.tsx:
```diff
@@ -1,9 +1,13 @@
 import styled from 'styled-components';
 
+import 'react-daypicker/lib/DayPicker.css';
+import DayPicker from 'react-daypicker';
```
These imports aren't used

NickCampagna commented:
> I see what you're going for here, and this is a good start to react. But I definitely think you're over complicating this. Look at https://www.npmjs.com/package/react-datepicker and follow their example and you'll have something that works completely and it'll be like ~20 lines of code.

There seems to be a **bunch** of different "datepicker"/"daypicker"/"day-picker"s out there, but that one seems to have the exact sort of functionality we needed by default - HUGE thanks for digging that up! I've implemented DatePicker, made it properly store the dueDate as a state and on update, it now properly calls calculateDaysLeft, so the "Days Left" text now updates whenever a new value is chosen. As a result, I've deleted the dueDateDropdown.tsx file and the large chunks of code that was previously in taskView.tsx supporting the triple drop-down method.
